cannon/mipsevm/tests/evm_multithreaded_test.go

```diff
@@ -594,52 +967,99 @@ func TestEVM_SysClockGettimeRealtime(t *testing.T) {
 func testEVM_SysClockGettime(t *testing.T, clkid uint32) {
 	var tracer *tracing.Hooks
 
+	llVariations := []struct {
+		name                   string
+		llReservationActive    bool
+		matchThreadId          bool
+		matchEffAddr           bool
+		matchEffAddr2          bool
+		shouldClearReservation bool
```

Comment:
nit: Don't need this. We can infer the value of `shouldClearReservation` from `matchEffAddr` and `matchEffAddr2`

Reply:
This check is pretty simple, so could probably go either way here, but I generally prefer to set explicit expectations in the test params vs. performing logic in the test to decide on the expectations.  I think you're slightly less likely to reproduce bugs in the test this way - forces you to think about the cases individually and reduces the chances of copying buggy logic from the implementation into the test.

Reply:
That's a good practice. Thanks for explaining.